core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/CarbonUtils.java

```diff
@@ -1358,4 +1368,63 @@ public static boolean isLegacyAuditLogsDisabled() {
 
         return Boolean.parseBoolean(System.getProperty(DISABLE_LEGACY_AUDIT_LOGS));
     }
+
+    /**
+     * Publish audit logs to the audit log file.
+     *
+     * @param auditLogData Audit log event data required for lpg.
+     */
+    public static void publishAuditLogs(Map<String, Object> auditLogData) {
+
+        String id = UUID.randomUUID().toString();
+        String recordedAt = String.valueOf(parseDateTime(Instant.now().toString()));
+        String clientComponent = auditLogData.get(CarbonConstants.LogEventConstants.CLIENT_COMPONENT) != null ?
+                auditLogData.get(CarbonConstants.LogEventConstants.CLIENT_COMPONENT).toString() : "null";
+        String correlationId = MDC.get(CORRELATION_ID_MDC) != null ? MDC.get(CORRELATION_ID_MDC).toString() : "null";
+        String initiatorId = auditLogData.get(CarbonConstants.LogEventConstants.INITIATOR_ID) != null ?
+                auditLogData.get(CarbonConstants.LogEventConstants.INITIATOR_ID).toString() : "null";
+        String initiatorName = auditLogData.get(CarbonConstants.LogEventConstants.INITIATOR_NAME) != null ?
+                auditLogData.get(CarbonConstants.LogEventConstants.INITIATOR_NAME).toString() : "null";
+        String initiatorType = auditLogData.get(CarbonConstants.LogEventConstants.INITIATOR_TYPE) != null ?
+                auditLogData.get(CarbonConstants.LogEventConstants.INITIATOR_TYPE).toString() : "null";
+        String eventType = auditLogData.get(CarbonConstants.LogEventConstants.EVENT_TYPE) != null ?
+                auditLogData.get(CarbonConstants.LogEventConstants.EVENT_TYPE).toString() : "null";
+        String targetId = auditLogData.get(CarbonConstants.LogEventConstants.TARGET_ID) != null ?
+                auditLogData.get(CarbonConstants.LogEventConstants.TARGET_ID).toString() : "null";
+        String targetName = auditLogData.get(CarbonConstants.LogEventConstants.TARGET_NAME) != null ?
+                auditLogData.get(CarbonConstants.LogEventConstants.TARGET_NAME).toString() : null;
```

Comment:
shouldn't this be "null" if object is null ?

Reply:
also its discouraged to use the ternary operator since it affects code readability.

Reply:
resolved in e61201e